### ISSUE TITLE
Hide noisy variables in go-to-symbol palette

### ISSUE THREAD

**Issue by maxrothman** [microsoft/vscode#86761]
It's sometimes difficult to use "Go To Symbol" to go to a specific method of a class because the namespace of the palette is polluted with variable names. Personally, I have no need to use "Go To Symbol" to navigate to variables, so just like I can hide them in the outline, I'd like to be able to hide them from the palette.

**Comment by circuuz** [comment]
I fully agree. I feel like it would make sense to have settings similar to the outline view.

**Comment by bpasero** [comment]
We do have a simple check for excluding local symbols here:

https://github.com/microsoft/vscode/blob/798481c7978af034b054e14ace4e471e7a83f665/src/vs/workbench/contrib/search/browser/openSymbolHandler.ts#L228

It would be relatively easy to have this for the new quick open, but I am uncertain:
* should this be another quick open besides the "#" we have (or maybe you could type "`#:`" to do this flip without having to use an entire different prefix)?
* should this just be a setting

And then, when we merge file and symbol results (when `search.quickOpen.includeSymbols` is enabled), we actually drop the local symbols which some people also do not like (see https://github.com/microsoft/vscode/issues/52773).

**Comment by bpasero** [comment]
I added a new setting for this purpose:

`search.quickOpen.workspaceSymbolsFilter`: Controls the filter to apply for the workspace symbols search in quick open. Depending on the setting, some symbols like local variables will be excluded to reduce the total number of results
* `default`: All symbols including local variables are included in the specific workspace symbols picker but excluded from the files picker when `#search.quickOpen.includeSymbols#` is enabled.
* `reduced`: Some symbols like local variables are excluded in all pickers
* `all`: All symbols including local variables are included in all pickers

@maxrothman @circuuz to clarify, do you talk about the workspace symbols picker or the local editor symbols picker? We could expand this setting to the local editor symbols picker too I guess.

**Comment by maxrothman** [comment]
I was originally looking for this to apply to the local editor symbols picker (cmd+shift+o), but I suppose it'd be useful in the workspace symbols picker (cmd+t) as well.

I would strongly advocate for the settings to simply be a duplicate of the outline ones but applied to the palettes, both for consistency's sake, and to give users the power to configure things the way they want them.

**Comment by bpasero** [comment]
@maxrothman one challenge is that we have 3 locations where symbols can appear in quick open:
* goto symbol (editor scoped)
* open symbol (workspace soped)
* open file & symbol (when `search.quickOpen.includeSymbols` is enabled)

Having one setting per symbol kind multiplied by 3 for each of those locations would mean a lot of settings to add...

**Comment by maxrothman** [comment]
Instead of having n quick open methods x m settings, why not have m settings that each take an object? Here's an example:

```js
// Show `array`-symbols in all 3 places
"goto.showArrays": true,

// Don't show `boolean`-symbols in any place
"goto.showBooleans": false,

// Show `class`-symbols only in the editor-scoped goto-symbol palette
"goto.showClasses": {
    "editor": true
},

// And so on...
"goto.showConstants": {
    "quickOpen": true,
    "workspace": true
},
```
It would be nice to eventually unify this with the outline settings, since I suspect most users usually want to enable/disable symbols for all locations most of the time, but that ship has unfortunately already sailed.

**Comment by circuuz** [comment]
@bpasero I was talking about the local editor symbols picker, but it would of course be nice to have the same settings for both the local editor symbols picker and the workspace symbol picker.

Regarding how to set up the settings I also believe that most users using these settings would use the same settings for all the different locations/pickers.

**Comment by bpasero** [comment]
I removed the new setting (from https://github.com/microsoft/vscode/issues/86761#issuecomment-601102253) given there is probably more required. May not happen for March though.

**Comment by vict0rsch** [comment]
Any ETA on this? I posted a StackOverflow question without realizing it may be (is?) related to this issue

I'm using `Go To Symbols in Workspace...` to navigate functions/methods and I don't really care about variables. Is there a setting I can change to get this behaviour? 

Similar to `Outline: Show Variables` being turned off to ignore variable in the outline

[![vscode go to symbols][1]][1]


  [1]: https://i.stack.imgur.com/3KmCa.png

**Comment by daniel-schmidt** [comment]
It would be great to have an option to navigate all classes or functions in the workspace. The current version showing just every symbol gets quite bloated in larger code bases. A global option in the settings is already useful, but I don't want it to influence the editor-scoped picker as well, since in this narrow scope, I usually care about everything in the editor. But in general, a dynamic filter option or another kind of quick open with a different symbol than # and the possibility to assign a different keyboard shortcut would be much more useful for me.

**Comment by ian-h-chamberlain** [comment]
Similar to the above comment, it would be great to filter by type in a single search, e.g. something like `@:method foo`, `@:var bar` (or a new sigil like `@@` or something, if needed). Being able to group by type with `@:` is useful, but it still requires a fair amount of navigation in the quick picker if e.g. classes or free functions (sorted first) share a name with a method or variable that you are looking for.

I am not sure if having a per-search by-type filter like this would satisfy the OP's use case, but if it were implemented it would probably have to interact with that kind of configuration (override it? intersect filtering? not sure)..
